Consider replacing AppVeyor with Azure Pipelines

We seem to have too little capacity on AppVeyor as CI builds are often queued for a long time. Azure Pipelines apparently supports Windows and 10 parallel jobs, for free. It might be a better alternative for mypy, since our builds are somewhat slow.

Is anybody interested in looking at this? Any help would be appreciated, even just small experiments and helpful pointers.

Here's a blog post that talks about the open source offering: https://azure.microsoft.com/en-gb/blog/announcing-azure-pipelines-with-unlimited-ci-cd-minutes-for-open-source/

```
2020-01-17T21:42:49.9175964Z ================================== FAILURES ===================================
2020-01-17T21:42:49.9176855Z ________________________________ testDictFree _________________________________
2020-01-17T21:42:49.9177779Z [gw1] win32 -- Python 3.7.6 D:\a\mypy\mypy\.tox\py37\Scripts\python.EXE
2020-01-17T21:42:49.9178633Z data: D:\a\mypy\mypy\mypyc\test-data\run.test:4408:
2020-01-17T21:42:49.9179149Z D:\a\mypy\mypy\mypyc\test\test_run.py:115: in run_case
2020-01-17T21:42:49.9179442Z     self.run_case_inner(testcase)
2020-01-17T21:42:49.9179699Z D:\a\mypy\mypy\mypyc\test\test_run.py:130: in run_case_inner
2020-01-17T21:42:49.9179960Z     self.run_case_step(testcase, step)
2020-01-17T21:42:49.9180936Z D:\a\mypy\mypy\mypyc\test\test_run.py:286: in run_case_step
2020-01-17T21:42:49.9181261Z     assert_test_output(testcase, outlines, msg, expected)
2020-01-17T21:42:49.9181562Z D:\a\mypy\mypy\mypyc\test\testutil.py:155: in assert_test_output
2020-01-17T21:42:49.9181851Z     '{} ({}, line {})'.format(message, testcase.file, testcase.line))
2020-01-17T21:42:49.9182172Z D:\a\mypy\mypy\mypy\test\helpers.py:116: in assert_string_arrays_equal
2020-01-17T21:42:49.9182450Z     raise AssertionError(msg)
2020-01-17T21:42:49.9182822Z E   AssertionError: Invalid output (D:\a\mypy\mypy\mypyc\test-data\run.test, line 4408)
2020-01-17T21:42:49.9183129Z ---------------------------- Captured stdout call -----------------------------
2020-01-17T21:42:49.9183404Z running build_ext
2020-01-17T21:42:49.9183647Z building 'native' extension
2020-01-17T21:42:49.9183898Z creating build\temp.win-amd64-3.7
2020-01-17T21:42:49.9184099Z creating build\temp.win-amd64-3.7\Release
2020-01-17T21:42:49.9184362Z creating build\temp.win-amd64-3.7\Release\build
2020-01-17T21:42:49.9184901Z C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.24.28314\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -ID:\a\mypy\mypy\mypyc\lib-rt -Ic:\hostedtoolcache\windows\python\3.7.6\x64\include -Ic:\hostedtoolcache\windows\python\3.7.6\x64\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.24.28314\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.24.28314\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcbuild\__native.c /Fobuild\temp.win-amd64-3.7\Release\build\__native.obj /O2 /wd4102 /wd4101 /wd4146
2020-01-17T21:42:49.9185340Z __native.c
2020-01-17T21:42:49.9185493Z 
2020-01-17T21:42:49.9186199Z D:\a\mypy\mypy\mypyc\lib-rt\getargs.c(1432): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
2020-01-17T21:42:49.9186453Z 
2020-01-17T21:42:49.9186718Z creating C:\Users\RUNNER~1\AppData\Local\Temp\mypy-test-rmu38wl8\tmp\build\lib.win-amd64-3.7
2020-01-17T21:42:49.9187393Z C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.24.28314\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\hostedtoolcache\windows\python\3.7.6\x64\Libs /LIBPATH:D:\a\mypy\mypy\.tox\py37\libs /LIBPATH:D:\a\mypy\mypy\.tox\py37\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.24.28314\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.24.28314\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\um\x64" /EXPORT:PyInit_native build\temp.win-amd64-3.7\Release\build\__native.obj /OUT:build\lib.win-amd64-3.7\native.cp37-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.7\Release\build\native.cp37-win_amd64.lib
2020-01-17T21:42:49.9187803Z    Creating library build\temp.win-amd64-3.7\Release\build\native.cp37-win_amd64.lib and object build\temp.win-amd64-3.7\Release\build\native.cp37-win_amd64.exp
2020-01-17T21:42:49.9188064Z 
2020-01-17T21:42:49.9188321Z Generating code
2020-01-17T21:42:49.9188479Z 
2020-01-17T21:42:49.9188729Z Finished generating code
2020-01-17T21:42:49.9188889Z 
2020-01-17T21:42:49.9189182Z copying build\lib.win-amd64-3.7\native.cp37-win_amd64.pyd -> 
2020-01-17T21:42:49.9189344Z 
2020-01-17T21:42:49.9189592Z *** Exit status: 1
2020-01-17T21:42:49.9190441Z ---------------------------- Captured stderr call -----------------------------
2020-01-17T21:42:49.9190761Z Expected:
2020-01-17T21:42:49.9191037Z Actual:
2020-01-17T21:42:49.9191306Z   Traceback (most recent call last):            (diff)
2020-01-17T21:42:49.9195768Z     File "driver.py", line 12, in <module>      (diff)
2020-01-17T21:42:49.9196613Z       test()                                    (diff)
2020-01-17T21:42:49.9197328Z     File "driver.py", line 10, in test          (diff)
2020-01-17T21:42:49.9197630Z       assert x == y                             (diff)
2020-01-17T21:42:49.9197839Z   AssertionError                                (diff)
2020-01-17T21:42:49.9198123Z 
```

any hints on this one? it's the only remaining failing test

my guess is this is an actual failure -- looks like it's crashing the compiler on this newer version of visual studio?

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -1,0 +1,49 @@
+name: main
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+  pull_request:
+    paths-ignore:
+    - 'docs/**'
+    - '**/*.rst'
+    - '**/*.md'
+    - .gitignore
+    - .travis.yml
+    - CREDITS
+    - LICENSE
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [windows-py37-32, windows-py37-64]
+        include:
+        - name: windows-py37-32
+          python: '3.7'
+          arch: x86
+          os: windows-latest
+          toxenv: py37
+        - name: windows-py37-64
+          python: '3.7'
+          arch: x64
+          os: windows-latest
+          toxenv: py37
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: initialize submodules
+      run: git submodule update --init
+    - uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python }}
+        architecture: ${{ matrix.arch }}
+    - name: install tox
+      run: pip install --upgrade setuptools tox==3.9.0
+    - name: setup tox environment
+      run: tox -e ${{ matrix.toxenv }} --notest
+    - name: test
+      run: tox -e ${{ matrix.toxenv }}

--- a/tox.ini
+++ b/tox.ini
@@ -13,7 +13,7 @@ isolated_build = true
 [testenv]
 description = run the test driver with {basepython}
 setenv = cov: COVERAGE_FILE={toxworkdir}/.coverage.{envname}
-passenv = PYTEST_XDIST_WORKER_COUNT
+passenv = PYTEST_XDIST_WORKER_COUNT PROGRAMDATA PROGRAMFILES(X86)
 deps = -rtest-requirements.txt
 commands = python -m pytest {posargs}
            cov: python -m pytest {posargs: --cov mypy --cov-config setup.cfg}

